### PR TITLE
Fixed Sonarcloud warning.

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6411572&bid=20777980)

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java [hunk 1]
```diff
@@ -119,10 +119,8 @@ public boolean handleRequest(
                 classicCallback.invoke(rootNode);
                 return true;
             }
-        } catch (JsonMappingException e) {
-            e.printStackTrace();
         } catch (JsonProcessingException e) {
-            e.printStackTrace();
+            LOG.warn("Caching issue during weight check." + e.getMessage(), e);
```
Comment (wcekan) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java:123]: Perhaps
```suggestion
            LOG.warn("Caching issue during weight check. {}", e.getMessage(), e);
```

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6411572&bid=20777980)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6411572&bid=20777980)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6411572&bid=20777980)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=6411572&bid=20777980)

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java [hunk 2]
```diff
@@ -84,7 +82,7 @@ public boolean handleRequest(
         try {
             LOG.debug("Weight query {}", writer.writeValueAsString(weightEvaluationQuery));
```
Comment (wcekan) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java:83]:
```suggestion
            if (LOG.isDebugEnabled()) {
                LOG.debug("Weight query {}", writer.writeValueAsString(weightEvaluationQuery));
            }
```

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java [hunk 3]
```diff
@@ -148,17 +144,14 @@ protected SuccessCallback buildCacheSuccessCallback(
...
-                    LOG.warn("Cache write json exception:", e);
-                }
+        return jsonResult -> {
+            // send the response to the user before waiting to cache
+            successCallback.invoke(jsonResult);
+            try {
+                querySignedCacheService.writeCache(response, jsonResult, druidQuery);
+            } catch (JsonProcessingException e) {
+                // Warn on cache write exception only
+                LOG.warn(String.format("Caching writing exception. %s", e.getMessage()), e);
```
Comment (wcekan) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java:154]: Is this better than using the LOG formatting?
```suggestion
                LOG.warn("Caching writing exception. {}", e.getMessage(), e);
```

**Review by sgoplani — APPROVED** [review]

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java (hunk 3 above)
Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java:156]: I wanted to take advantage of the logger option that expects throwable. Those two syntaxes (syntactes?) (variable substitution and throwable) are incompatible.

Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandler.java:156]: and the whole point of this block that we're fixing is to capture the trace on issues.

**Pull request merged**